cruise-control-core/src/main/java/com/linkedin/cruisecontrol/monitor/sampling/aggregator/MetricValues.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling.aggregator;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.StringJoiner;
+
+
+/**
+ * A class hosting the values of a particular metric.
+ */
+public class MetricValues implements Serializable {
+  private static final long serialVersionUID = -6840253566423285966L;
+  
+  private final float[] _values;
+  private volatile float _max;
+  private volatile float _sumForAvg;
+
+  public MetricValues(int numWindows) {
+    _values = new float[numWindows];
+    _sumForAvg = 0;
+    _max = 0;
+  }
+
+  public void set(int index, double value) {
+    if (_max == _values[index] && value < _max) {
+      _max = -1.0f;
+    }
+    _sumForAvg += value - _values[index];
+    _values[index] = (float) value;
+  }
+
+  public double get(int index) {
+    return _values[index];
+  }
+  
+  public void clear() {
+    for (int i = 0; i < _values.length; i++) {
+      _values[i] = 0;
+    }
+    _sumForAvg = 0;
+    _max = 0;
+  }
+
+  public int length() {
+    return _values.length;
+  }
+  
+  public void add(double[] values) {
+    _max = 0;
```

Comment:
If this gets initialized to something like -Double.MIN_VALUE then this can be made safe for metrics that can assume negative values.  I know this is not common, but once this is in the wild it's difficult to say how this will get used.